src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -85,10 +85,20 @@ export class MissingTerms extends Component {
    *Returns all original basic query expression terms that were not matched by the result item the component instance is associated with.
    */
   public get missingTerms(): string[] {
-    return this.result.absentTerms.filter(term => {
+    const terms = [];
+
+    for (const term of this.result.absentTerms) {
       const regex = this.createWordBoundaryDelimitedRegex(term);
-      return regex.test(this.queryStateModel.get('q'));
-    });
+      const query = this.queryStateModel.get('q');
+      const result = regex.exec(query);
+
+      if (result) {
+        const termInQuery = result[4];
```

Comment:
Instead of `result[4]`, could we have better naming about why we want to use this capture group specifically ? (the original keyword?). 

Or some comment that explains what `result[4]` is ?

Reply:
I like your suggestion of mentioning "original keyword". I prefer having a well-named variable versus leaving a comment. I will change the name to: `originalKeywordInQuery`.